frontend/src/components/Schedule/Schedule.tsx

```diff
@@ -20,9 +39,43 @@ const Schedule = () => {
   const defaultEnd = new Date(defaultStart.getTime() + 28699999);
 
   const [curStart, setCurStart] = useState(defaultStart);
-  const [events, setEvents] = useState(tempEvents);
+  const [events, setEvents] = useState<CalEvent[]>([]);
+  const [drivers, setDrivers] = useState<Driver[]>([]);
   const [viewState, setviewState] = useState(false);
 
+  useEffect(() => {
+    fetch('/rides')
+      .then((res) => res.json())
+      .then(({ data }) => {
+        console.log(data)
+        setEvents(data
+          .filter((ride: any) => ride.type !== "unscheduled" && ride.driver !== undefined)
```

Comment:
types are located in `components/types/index.ts`, use the ones in there instead of `any`
also a ride can be a past ride, do we still want to display those ones?

Reply:
for the type because ride's `driver` can be undefined, setting its type to `Ride`will cause red lines. for the past rides, I'm thinking we want the previous rides in that day still on the calendar (but I might be wrong)

Reply:
yeah that makes sense for past rides
i see the issue, i checked and it doesn't look like your branch has the updated types for `Ride` which should take care of the typing issues. try updating your branch from master and see if that resolves the issue